plugins/aws/connect/voiceIdDomainEncrypted.js

```diff
@@ -0,0 +1,104 @@
+    settings: {
+        voice_id_desired_encryption_level: {
+            name: 'Connect Voice ID Domain Encrypted',
+            description: 'In order (lowest to highest) awscmk=Customer managed KMS; externalcmk=Customer managed externally sourced KMS; cloudhsm=Customer managed CloudHSM sourced KMS',
+            regex: '^(awscmk|externalcmk|cloudhsm)$',
+            default: 'awscmk'
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        var config = {
+            desiredEncryptionLevelString: settings.voice_id_desired_encryption_level || this.settings.voice_id_desired_encryption_level.default
+        };
+
+        var desiredEncryptionLevel = helpers.ENCRYPTION_LEVELS.indexOf(config.desiredEncryptionLevelString);
+        var currentEncryptionLevel;
+
+        async.each(regions.voiceid, function(region, rcb){
+            var listDomains = helpers.addSource(cache, source,
+                ['voiceid', 'listDomains', region]);
+
+            if (!listDomains) return rcb();
+
+            if (listDomains.err || !listDomains.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query voiceid domain: ' + helpers.addError(listDomains), region);
+                return rcb();
+            }
+
+            if (!listDomains.data.length) {
+                helpers.addResult(results, 0, 'No voiceid domain found', region);
+                return rcb();
+            }
+
+            var listKeys = helpers.addSource(cache, source,
+                ['kms', 'listKeys', region]);
+
+            if (!listKeys || listKeys.err || !listKeys.data) {
+                helpers.addResult(results, 3,
+                    `Unable to list KMS keys: ${helpers.addError(listKeys)}`, region);
+                return rcb();
+            }
+
+            for (let domain of listDomains.data) {
+                let resource = domain.Arn;
+
+                if (domain.ServerSideEncryptionConfiguration.KmsKeyId) {
+                    let encryptionKey = domain.ServerSideEncryptionConfiguration.KmsKeyId;
+                    var keyId = encryptionKey.split('/')[1] ? encryptionKey.split('/')[1] : encryptionKey;
+
+                    var describeKey = helpers.addSource(cache, source,
+                        ['kms', 'describeKey', region, keyId]);
+
+                    if (!describeKey || describeKey.err || !describeKey.data || !describeKey.data.KeyMetadata) {
+                        helpers.addResult(results, 3,
+                            `Unable to query KMS key: ${helpers.addError(describeKey)}`,
+                            region, encryptionKey);
+                        continue;
+                    }
+
+                    currentEncryptionLevel = helpers.getEncryptionLevel(describeKey.data.KeyMetadata, helpers.ENCRYPTION_LEVELS);
+                } else {
+                    helpers.addResult(results, 3,
```

Comment:
We are saying that its encrypted by default so current encryption level will be 2

Reply:
so we will compare here as well and see if the desired level is reached then successful result otherwise fail right?